frontend/src/v5/ui/routes/dashboard/projects/settingsForm/settingsForm.component.tsx

```diff
@@ -0,0 +1,335 @@
+	formatMessage({ id: 'category.type.gis', defaultMessage: 'GIS' }),
+	formatMessage({ id: 'category.type.infrastructure', defaultMessage: 'Infrastructure' }),
+	formatMessage({ id: 'category.type.interior', defaultMessage: 'Interior' }),
+	formatMessage({ id: 'category.type.ladscape', defaultMessage: 'Landscape' }),
+	formatMessage({ id: 'category.type.mep', defaultMessage: 'MEP' }),
+	formatMessage({ id: 'category.type.mechanical', defaultMessage: 'Mechanical' }),
+	formatMessage({ id: 'category.type.structural', defaultMessage: 'Structural' }),
+	formatMessage({ id: 'category.type.survey', defaultMessage: 'Survey' }),
+	formatMessage({ id: 'category.type.other', defaultMessage: 'Other' }),
+];
+
+interface IFormInput {
+	name: string;
+	unit: string;
+	desc: string;
+	code: string;
+	category?: string,
+	defaultView: string;
+	latitude: number;
+	longitude: number;
+	angleFromNorth: number;
+	x: number;
+	y: number;
+	z: number;
+}
+
+const getDefaultValues = (containerOrFederation: IContainer | IFederation, isContainer?: boolean) => {
+	const DEFAULT_UNIT = UNITS[0];
+	const {
+		unit = DEFAULT_UNIT.abbreviation,
+		angleFromNorth = 0,
+		code,
+		name,
+		desc = '',
+	} = containerOrFederation;
+	const defaultView = containerOrFederation.defaultView || EMPTY_VIEW._id;
+	const { latLong, position } = containerOrFederation.surveyPoint || {};
+	const [x, y, z] = position || [0, 0, 0];
+	const [latitude, longitude] = latLong || [0, 0];
+	return {
+		name,
+		desc,
+		code,
+		unit,
+		latitude,
+		longitude,
+		angleFromNorth,
+		defaultView,
+		x,
+		y,
+		z,
+		...(isContainer ? { category: containerOrFederation.category || 'Uncategorised' } : {}),
+	};
+};
+
+type ISettingsForm = {
+	open: boolean;
+	containerOrFederation: IContainer | IFederation;
+	settingsSchema: any;
+	isContainer?: boolean;
+	onClose: () => void;
+	fetchSettings: (teamspace: string, project: string, containerOrFederationId: string) => void;
+	fetchViews: (teamspace: string, project: string, containerOrFederationId: string) => void;
+	updateSettings: (
+		teamspace: string,
+		project: string,
+		containerOrFederationId: string,
+		settings: ContainerSettings | FederationSettings,
+	) => void;
+};
+
+export const SettingsForm = ({
+	open,
+	containerOrFederation,
+	settingsSchema,
+	isContainer,
+	fetchSettings,
+	fetchViews,
+	updateSettings,
+	onClose,
+}: ISettingsForm) => {
+	let defaultValues = getDefaultValues(containerOrFederation, isContainer) as any;
+	const {
+		handleSubmit,
+		reset,
+		watch,
+		control,
+		formState,
+		formState: { errors },
+	} = useForm<IFormInput>({
+		mode: 'onChange',
+		resolver: yupResolver(settingsSchema),
+		defaultValues,
+	});
+
+	const currentUnit = watch('unit');
+
+	const { teamspace, project } = useParams() as { teamspace: string, project: string };
+
+	const resetValues = () => {
+		defaultValues = getDefaultValues(containerOrFederation, isContainer) as any;
```

Comment:
I might be understanding this wrong, but won't "defaultValues" never change? If so it should be a const and removed from this function

Reply:
fixed